apps/supabase-auth-service/supabase_auth_service/app.py

```diff
@@ -0,0 +1,122 @@
+
+
+@app.route("/auth/v1/verify", methods=["POST"])
+def verify():
+    try:
+        data = request.get_json()
+        res = supabase.auth.verify_otp(
+            {"email": data["email"], "token": data["token"], "type": "signup"}
+        )
+        return jsonify(
+            {
+                "data": {
+                    "access_token": res.session.access_token,
+                    "refresh_token": res.session.refresh_token,
+                },
+                "meta": {"status": 200},
+            }
+        )
+    except Exception as e:
+        return jsonify({"data": None, "meta": {"status": 400, "message": str(e)}})
+
+
+@app.route("/auth/v1/sign_in", methods=["POST"])
+def sign_in():
+    try:
+        data = request.get_json()
+        res = supabase.auth.sign_in_with_password(
+            {"email": data["email"], "password": data["password"]}
+        )
+        return jsonify(
+            {
+                "data": {
+                    "access_token": res.session.access_token,
+                    "refresh_token": res.session.refresh_token,
+                },
+                "meta": {"status": 200},
+            }
+        )
+    except Exception as e:
+        return jsonify({"data": None, "meta": {"status": 400, "message": str(e)}})
+
+
+@app.route("/auth/v1/refresh_access_token", methods=["POST"])
+def refresh_session():
+    try:
+        data = request.get_json()
+        res = supabase.auth._refresh_access_token(data["refresh_token"])
+        return jsonify(
+            {
+                "data": {
+                    "access_token": res.session.access_token,
+                    "refresh_token": res.session.refresh_token,
+                },
+                "meta": {"status": 200},
+            }
+        )
+    except Exception as e:
+        return jsonify({"data": None, "meta": {"status": 400, "message": str(e)}})
+
+
+@app.route("/auth/v1/reset_password", methods=["POST"])
+def reset_password():
+    try:
+        data = request.get_json()
+        supabase.auth.reset_password_email(
+            data["email"]
+        )  # doesn't return anything, so just pray i guess 🤷‍♂️
+        return jsonify({"data": None, "meta": {"status": 200}})
+    except Exception as e:
+        return jsonify({"data": None, "meta": {"status": 400, "message": str(e)}})
+
+
+@app.route("/auth/v1/sign_out", methods=["POST"])
+def sign_out():
+    try:
+        supabase.auth.sign_out()
+        return jsonify({"data": None, "meta": {"status": 200}})
+    except Exception as e:
+        return jsonify({"data": None, "meta": {"status": 400, "message": str(e)}})
+
+
+@app.route("/auth/v1/users/<string:user_id>/metadata", methods=["GET"])
+def get_user(user_id):
+    try:
+        # Add check to see if user is admin
+        # if user.isadmin:
+        res = supabase.auth.admin.get_user_by_id(user_id)
+        filtered_dict = {
+            k: v for k, v in res.user.__dict__.items() if k != "identities"
```

Comment:
seems like you can do smth like this to be pythonic because it has a `__dict__` dunder method:

```suggestion
            k: v for k, v in res.user.items() if k != "identities"
```

Reply:
but the user object has no items() method? is there an alternative method you want me to use to convert from user object to  json format?

Reply:
use `dict()` instead
```suggestion
            k: v for k, v in res.user.dict().items() if k != "identities"
```